nix/installer-bootstrap/installer-configuration.nix

```diff
@@ -65,11 +65,15 @@
     pkgs.gptfdisk
     pkgs.parted
     pkgs.cryptsetup
+    pkgs.wget
+    pkgs.curl
+    pkgs.magic-wormhole
   ];
 
-  # save space and compilation time. might revise?
-  hardware.enableAllFirmware = lib.mkForce false;
-  hardware.enableRedistributableFirmware = lib.mkForce false;
+  # add in firmware so usb-ethernet adapters work
+  hardware.enableAllFirmware = lib.mkForce true;
+  hardware.enableRedistributableFirmware = lib.mkForce true;
+  nixpkgs.config.allowUnfree = true;
```

Comment:
Is allowing unfree software by default (e.g. by maintainers rather than an end-user) controversial or, worse, liability-causing? (Genuine question--I have no idea.)

Reply:
From the commit message:

"As booting Linux on Apple Silicon already requires many unfree Apple blobs, this change should not offend anyone's sensibilities."

Reply:
I would say `enableAllFirmware` and `allowUnfree` both need to be off. `enableAllFirmware` is specifically marked as including non-redistributable packages, and I don't want to expose the pre-hosted images to that liability. Additionally, it does not seem to include anything useful to us. This then makes `allowUnfree` unnecessary.